src/Compilers/CSharp/Portable/Symbols/Source/SourceTypeParameterSymbol.cs

```diff
@@ -171,12 +177,15 @@ internal override CustomAttributesBag<CSharpAttributeData> GetAttributesBag()
                 var sourceMethod = this.ContainingSymbol as SourceMemberMethodSymbol;
                 if ((object)sourceMethod == null || (object)sourceMethod.SourcePartialDefinition == null)
                 {
-                    lazyAttributesStored = LoadAndValidateAttributes(OneOrMany.Create(this.MergedAttributeDeclarationSyntaxLists), ref _lazyCustomAttributesBag);
+                    lazyAttributesStored = LoadAndValidateAttributes(
```

Comment:
@AlekseyTs I added your test and am working to fix it, but I don't understand how `GetDeclaredSymbol` is the problem here. It looks like diagnostics are only gathered in `BindStatement`, which is not called through any pathway in the semantic model. Shouldn't I force diagnostics in that code path as well (and then throw them away)?